project-1/main.py

```diff
@@ -4,21 +4,61 @@
 from selenium.webdriver.common.by import By
 from selenium.webdriver.chrome.service import Service as ChromeService
 from webdriver_manager.chrome import ChromeDriverManager
+import os, json
+from dotenv import load_dotenv, find_dotenv
+load_dotenv()
+KEY = json.loads(os.getenv("PROJECT_1202"))
+
 
 driver = webdriver.Chrome(service=ChromeService(ChromeDriverManager().install()))
 link = "https://www.food.com/ideas/top-comfort-food-recipes-6929#c-791290"
 driver.get(url=link)
 
-gc = gspread.service_account(filename='my-project-1202.json')
+gc = gspread.service_account(KEY)
+
+
+def open_spreadsheet(sheet_name):
+    """
+    This opens a spreadsheet already created on Google sheet
+    """
+    sh = gc.open(sheet_name)
+    logging.info("spreadsheet opened")
+
+    def select_spreadsheet(s_name):
+        """
+        This function  selects the preferred worksheet
+        :param s_name: This is the name of the desired worksheet in str format
+        """
+        worksheet = sh.worksheet(s_name)
+        return worksheet
+    return select_spreadsheet
+
+
+def create_spreadsheet(sheet_name):
+    """
+    This creates a new spreadsheet file in google sheet
+    """
+    sh = gc.create(sheet_name)
+    logging.info("Spreadsheet created")
 
-sh = gc.open("project 1202")
+    def select_spreadsheet(s_name):
```

Comment:
You shouldn't nest this function

Reply:
hi @tobaadesugba i nested this function because i would be needing the sh variable which i cant access if it is in the local scope of the function

Reply:
i deleted a comment of your by mistake

Reply:
Hi @tobaadesugba  i have made the changes please review

Reply:
> i deleted a comment of your by mistake

No worries, it was still referring to the same issue